tests/cpp/test_multidevice_overlap.cpp

```diff
@@ -1035,10 +1053,31 @@ class RingAllgatherOverlapTest : public MultiDeviceTest {
     auto tc_unsharded_expected_ =
         torch::matmul(ta_unsharded_, tb_unsharded_.cpu());
     EXPECT_TRUE(
-        tc_unsharded_.cpu().allclose(tc_unsharded_expected_, 1e-1, 1e-1))
+        tc_unsharded_.cpu().allclose(tc_unsharded_expected_, 5e-1, 5e-1))
         << "Unexpected results, obtained: " << tc_unsharded_
         << "expected: " << tc_unsharded_expected_;
   }
+
+  void startCounter() {
+    auto stream = at::cuda::getCurrentCUDAStream();
+    C10_CUDA_CHECK(cudaEventRecord(start_event_, stream));
+  }
+
+  void addCounter() {
```

Comment:
Congratulations! You reinvented https://google.github.io/benchmark/user_guide.html#manual-timing

With some refactoring similar to https://github.com/NVIDIA/Fuser/blob/d2f333f1284f00e302ff97563e02765257d80893/CMakeLists.txt#L932-L933, I think we can split the numerical test and the performance benchmark. This way, the numerical test can use small sizes and integers for stability. The benchmark tests can use large sizes and at::randn for https://www.thonking.ai/p/strangely-matrix-multiplications, and measure the timing with https://google.github.io/benchmark/user_guide.html#manual-timing